django_coverage_plugin/plugin.py

```diff
@@ -43,28 +42,24 @@ def check_debug():
     to do its work.  Check that the setting is correct, and raise an exception
     if it is not.
 
+    Returns True if the debug check was performed, False otherwise
     """
-    # The settings for templates changed in Django 1.8 from TEMPLATE_DEBUG to
-    # TEMPLATES[..]['debug'].  Django 1.9 tolerated both forms, 1.10 insists on
-    # the new form.  Don't try to be version-specific here.  If the new
-    # settings exist, use them, otherwise use the old.
-
     from django.conf import settings
 
-    try:
-        templates = getattr(settings, 'TEMPLATES', [])
-    except ImproperlyConfigured:
-        # Maybe there are no settings at all.  We are fine with this.  Our
-        # code will need settings, but if this program we're in runs without
-        # settings, then it must be that it never uses templates, and our code
-        # will never try to use the settings anyway.
-        return
-
-    if templates:
-        for template_settings in templates:
-            if template_settings['BACKEND'] != 'django.template.backends.django.DjangoTemplates':
-                raise DjangoTemplatePluginException("Can't use non-Django templates.")
-            if not template_settings.get('OPTIONS', {}).get('debug', False):
+    if not settings.configured:
```

Comment:
Here's how we defer check_debug until settings are configured.  Note that we can't really encounter the case where we'd be tracing a template until after settings have been configured because the template can't be run without configured settings.